src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -39,6 +66,36 @@ public EnumGroupAndName(string group, string name)
         /// <summary>
         /// Gets the name.
         /// </summary>
-        public string Name { get; }
+        public string Name
+        {
+            get
+            {
+                return _name();
+            }
+        }
+
+        // Equals and GetHashCode must be overloaded to accommodate the _nameFunc
+        public override bool Equals(object obj)
+        {
+            if (!(obj is EnumGroupAndName))
+            {
+                return false;
+            }
+
+            var second = (EnumGroupAndName)obj;
+
+            return string.Equals(Group, second.Group, StringComparison.Ordinal)
+                && _name == second._name;
```

Comment:
`Func` equality is not a straightforward thing. But is `EnumGroupAndName` equality important anywhere? I know we create `IList<KeyValuePair<EnumGroupAndName, string>>` collections but does any product code require a meaningful `Equals()` e.g. does the product code use any `IDictionary<EnumGroupAndName, string>` instances?

I'd rather `EnumGroupAndName` avoid overriding `Equals()` and `GetHashCode()` if possible. Fine to create a test-only `EnumGroupAndNameComparer` which uses the `Group` and `Name` properties (perhaps in `CultureInfo.InvariantCulture` to be consistent).